src/compiler/transformers/decorators-to-static/convert-static-members.ts

```diff
@@ -13,6 +16,25 @@ export const hasStaticInitializerInClass = (classElm: ts.ClassElement): boolean
     ts.isPropertyDeclaration(classElm) &&
     classElm.initializer !== undefined &&
     Array.isArray(classElm.modifiers) &&
-    classElm.modifiers!.some((modifier) => modifier.kind === ts.SyntaxKind.StaticKeyword)
+    classElm.modifiers!.some((modifier) => modifier.kind === ts.SyntaxKind.StaticKeyword) &&
+    !classElm.modifiers!.some(isStencilStateOrPropDecorator)
   );
 };
+
+/**
+ * Determines is a Modifier-like node is a Stencil `@Prop()` or `@State()` decorator
+ * @param modifier the AST node to evaluate
+ * @returns true if the node is a decorator with the name 'Prop' or 'State', false otherwise
+ */
+const isStencilStateOrPropDecorator = (modifier: ts.ModifierLike): boolean => {
+  if (ts.isDecorator(modifier)) {
+    const decoratorName =
+      ts.isCallExpression(modifier.expression) &&
+      ts.isIdentifier(modifier.expression.expression) &&
+      modifier.expression.expression.text;
+    return (
+      typeof decoratorName !== 'boolean' && CONSTRUCTOR_DEFINED_MEMBER_DECORATORS.some((val) => val === decoratorName)
```

Comment:
are you doing the `.some(...)` stuff to get around the typing for `.includes` ? I created this dumb little [`readOnlyArrayHasStringMember`](https://github.com/ionic-team/stencil/blob/29ae12061984eadd46fe2a73a56a7245e4271cee/src/utils/util.ts#L234-L237) function which could be used for that purpose (it has to do a typecast :/ )